### ISSUE TITLE
C++ extension randomly gives an irrelevant error when using "auto" type parameters after adding/removing newlines

### ISSUE THREAD

**Issue by dyigitpolat** [microsoft/vscode-cpptools#6714]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Ubuntu-20.04
- VS Code Version: 1.52.1
- C/C++ Extension Version: 1.1.3
- Other extensions you installed (and if the issue persists after disabling them): C/C++ Extension pack, yes it persists.
- Does this issue involve using SSH remote to run the extension on a remote machine?: doesn't matter

C++ extension giving irrelevant error with compiling code

**Steps to reproduce**

Below code produces a C++ error but it can compile without any problems (g++):

```
#include <iostream>
#include <array>

void printArr(auto arr)
{
    for(const auto &elem : arr)
    {
        std::cout << elem << " ";
    }

    std::cout << std::endl;
}

int main()
{
    std::array<int, 5> myArray;
    myArray = { 0, 1, 2, 3, 4 }; 

    printArr(myArray);

    return 0;
}
```

the error:
"more than one instance of overloaded function \"printArr\" matches the argument list: -- function template \"void printArr(auto arr)\" (declared at line 5) -- function template \"int printArr(auto arr)\" (declared at line 5) -- argument types are: (std::array<int, 5UL>)"

**Expected behavior**

No error.

<details>
  <summary><strong>Logs</strong></summary>

```json
{
	"resource": "/home/yigit/code/test.cpp",
	"owner": "C/C++",
	"code": "308",
	"severity": 8,
	"message": "more than one instance of overloaded function \"printArr\" matches the argument list: -- function template \"void printArr(auto arr)\" (declared at line 5) -- function template \"int printArr(auto arr)\" (declared at line 5) -- argument types are: (std::array<int, 5UL>)",
	"source": "C/C++",
	"startLineNumber": 20,
	"startColumn": 5,
	"endLineNumber": 20,
	"endColumn": 13
}
```
</details>

**Screenshots**

**Additional context**

**Comment by sean-mcmanus** [comment]
I'm not reproing the issue on Ubuntu 20 with g++ 10.2 with c++20 -- can you provide more repro info?

**Comment by dyigitpolat** [comment]
> I'm not reproing the issue on Ubuntu 20 with g++ 10.2 with c++20 -- can you provide more repro info?

it does not directly show up, you may want to play with the code a bit (add/remove const keywords, duplicate some lines, hover over identifiers to see type info etc.). When I do these, the error eventually shows up.

**Comment by sean-mcmanus** [comment]
@dyigitpolat Okay, thanks -- I get a repro after adding some new lines. Our parser sometimes parses/updates code in a non-deterministic manner.

**Comment by sean-mcmanus** [comment]
I've reported a bug against our shared parser with VS at https://developercommunity.visualstudio.com/content/problem/1297949/c-intellisense-randomly-gives-an-error-after-addin.html

**Comment by michelleangela** [comment]
Updated issue title to better reflect root cause. It seems the IntelliSense error is caused by the usage of "auto" type in parameters. See issue #6934.

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0-insiders

**Comment by michelleangela** [comment]
Fix is released in the 1.3.0 release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0
